scio-smb/src/main/scala/com/spotify/scio/smb/syntax/SortMergeBucketSCollectionSyntax.scala

```diff
@@ -49,16 +53,20 @@ final class SortedBucketSCollection[T](private val self: SCollection[T]) {
    *   contains information about key function, bucket and shard size, etc.
    */
   @experimental
-  def saveAsSortedBucket(write: SortedBucketIO.Write[_, _, T]): ClosedTap[Nothing] = {
+  def saveAsSortedBucket(
+    write: SortedBucketIO.Write[_, _, T]
+  ): ClosedTap[T] = {
+    val beamValueCoder = self.internal.getCoder
+    implicit val valueCoder: Coder[T] = Coder.beam(beamValueCoder)
+
     if (self.context.isTest) {
-      val testOutput = TestDataManager.getOutput(self.context.testId.get)
-      testOutput[T](SortedBucketIOUtil.testId(write))(self)
+      TestDataManager.getOutput(self.context.testId.get)(SortedBucketIOUtil.testId(write))
+      ClosedTap(TapOf[T].saveForTest(self))
     } else {
-      self.applyInternal(write)
-    }
+      val writeResult = self.applyInternal(write)
 
-    // @Todo: Implement taps for metadata/bucket elements
-    ClosedTap[Nothing](EmptyTap)
+      ClosedTap(SmbIO.tap(write.getFileOperations, writeResult).apply(self.context))
```

Comment:
Skip apply in scala
```suggestion
      ClosedTap(SmbIO.tap(write.getFileOperations, writeResult)(self.context))
```

Reply:
unfortunately this doesn't work because `SmbIO::tap` requires an implicit coder (required by `bucketFiles.underlying.flatMap`)... so if a curried param is supplied, the compiler expects it to be that coder. We'd have to do:

```scala
ClosedTap(
   SmbIO.tap(write.getFileOperations, writeResult)(Coder[V])(self.context)
)
```